src/test/ui/pattern/issue-68393-let-pat-assoc-constant.stderr

```diff
@@ -0,0 +1,15 @@
+error[E0158]: associated consts cannot be referenced in patterns
+  --> $DIR/issue-68393-let-pat-assoc-constant.rs:20:40
+   |
+LL | pub fn test<A: Foo, B: Foo>(arg: EFoo, A::X: EFoo) {
+   |                                        ^^^^
+
+error[E0158]: associated consts cannot be referenced in patterns
+  --> $DIR/issue-68393-let-pat-assoc-constant.rs:22:9
+   |
+LL |     let A::X = arg;
+   |         ^^^^
```

Comment:
Can we customize the "patterns" wording in fn args and let bindings? I would expect people that are unfamiliar with the language to get confused by this. At least a `note` with something like (but better) "`let` statements introduce new bindings to some pattern, normally a simple name which are also patterns".

Reply:
I think this is unlikely that a beginner would write this code personally. Using an associated constant in an irrefutable context seems fairly niche. In fact, I found this ICE by trying to unify the code, realizing that a check was missing in the irrefutable contexts. `A::X` also does not seem confusable for a binding. I would prefer not to preempt better diagnostics here until a beginner shows up.

If we are going to improve diagnostics, I'd rather add "defined here" `span_label`s for `X` and `A`.